### PR TITLE
Fixing display publisher (Python 3)

### PR DESCRIPTION
**Pull request by henryiii**
This fixes the CapturingDisplayPublisher by matching the signature with the regular DisplayPublisher (bug surfaced in #10813).

### CODE REVIEW

**Comment by takluyver** [comment]
I suspect that `capture_output()` should also record the data passed as `transient=` and `update=`, rather than accepting them and discarding them. @minrk? In any case, making the signature valid is already an improvement over throwing an error.

**Comment by minrk** [comment]
@takluyver exactly right

**Comment by takluyver** [comment]
@henryiii do you want to have a go at that as part of this PR, now that it's no longer required for #10813? If you're not sure about how to do it, we can also just merge this and open an issue for storing the extra fields.

**Comment by henryiii** [comment]
Is there anywhere the captured info is processed (other than stderr and stdout)? It looks to be simple unless something uses the outputs and assumes it'a a two element list.

**Comment by takluyver** [comment]
As far as I can see, it's only published in one place, by `IPython.utils.capture.CapturedIO`.

File: IPython/utils/capture.py [hunk 1]
```diff
@@ -14,14 +14,17 @@
 
 
 class RichOutput(object):
-    def __init__(self, data=None, metadata=None):
+    def __init__(self, data=None, metadata=None, transient=None, update=False):
         self.data = data or {}
         self.metadata = metadata or {}
-    
+        self.transient = metadata or {}
```
Comment (takluyver) [inline IPython/utils/capture.py]: `transient or  {}`

**Comment by takluyver** [comment]
Other than the one note I made, this looks good to me.

File: IPython/utils/capture.py (hunk 1 above)
Comment (henryiii) [reply IPython/utils/capture.py]: Fixed

**Pull request merged**